src/pages/ExternalSupplierInvoicePage.js

```diff
@@ -0,0 +1,441 @@
+    const infoColumns = [
+      [
+        {
+          title: `${pageInfoStrings.entry_date}:`,
+          info: formatDate(transaction.entryDate),
+        },
+        {
+          title: `${pageInfoStrings.confirm_date}:`,
+          info: formatDate(transaction.confirmDate),
+        },
+        {
+          title: `${pageInfoStrings.total_price}:`,
+          info: this.state.totalPrice,
+        },
+      ],
+      [
+        {
+          title: `${pageInfoStrings.supplier}:`,
+          info: this.props.transaction.otherParty && this.props.transaction.otherParty.name,
+        },
+        {
+          title: `${pageInfoStrings.their_ref}:`,
+          info: transaction.theirRef,
+          onPress: this.openTheirRefEditor,
+          editableType: 'text',
+        },
+        {
+          title: `${pageInfoStrings.comment}:`,
+          info: transaction.comment,
+          onPress: this.openCommentEditor,
+          editableType: 'text',
+        },
+      ],
+    ];
+    return (
+      <PageInfo columns={infoColumns} isEditingDisabled={this.props.transaction.isFinalised} />
+    );
+  }
+
+  renderCell(key, transactionBatch) {
+    const isEditable = !this.props.transaction.isFinalised;
+    const type = isEditable ? 'editable' : 'text';
+    switch (key) {
+      default:
+        return transactionBatch[key];
+      case 'packSize': {
+        const renderedCell = {
+          type: type,
+          cellContents: String(transactionBatch.packSize),
+        };
+        return renderedCell;
+      }
+      case 'numberOfPacks': {
+        const renderedCell = {
+          type: type,
+          cellContents: String(transactionBatch.numberOfPacks),
+        };
+        return renderedCell;
+      }
+      case 'batch': {
+        const renderedCell = {
+          type: type,
+          cellContents: transactionBatch.batch,
+          keyboardType: 'default',
+        };
+        return renderedCell;
+      }
+      case 'costPrice': {
+        const renderedCell = {
+          type: type,
+          cellContents: String(transactionBatch.costPrice),
+        };
+        return renderedCell;
+      }
+      case 'expiryDate': {
+        const expiryDate = formatExpiryDate(transactionBatch.expiryDate);
+        const renderedCell = {
+          type: type,
+          cellContents: expiryDate || 'month/year',
+        };
+        return renderedCell;
+      }
+      case 'remove':
+        return {
+          type: 'checkable',
+          icon: 'md-remove-circle',
+          isDisabled: this.props.transaction.isFinalised,
+        };
+      // expiryDate
+    }
+  }
+
+  addNewSIitem(item) {
+    const { database, transaction } = this.props;
+    database.write(() => {
+      const transactionItem = createRecord(database, 'TransactionItem', transaction, item);
+      createRecord(
+        database,
+        'TransactionBatch',
+        transactionItem,
+        createRecord(database, 'ItemBatch', item, '')
```

Comment:
So this creates a new ItemBatch then the user edits and saves the `batch`(code) field; It's possible to have multiple entries for the same batch (given there may be a preexisting batch with the same `batch` string). @craigdrown or someone might want to confirm, but that is probably no deal.

I think it be better if there was a way to not create the corresponding itemBatches until finalise (where it can use the batch code entered by user. Finalising stock takes may have a clue on how to do this (and there is intent to make stock takes per batch, so maybe stock takes will reflect what we do here!).